### ISSUE TITLE
Embedded entities are always indexed

### ISSUE THREAD

**Issue by msomhorst** [objectify/objectify#397]
I am using Embedded Entities inside an entity, but I can't unindex those attributes: whatever I do, they are always indexed.

For example when I have this entity:
```
@Entity
private static class MyEntity {
    @Id Long id;
    EmbeddedEntity embeddedEntity;
    String title;
}
```
The `embeddedEntity` attribute is indexed, while the `title` is not. Adding a `@Unindex` annotation to the `embeddedEntity` or the `EmbeddedEntity` class does not help.

For completeness, this is how the EmbeddedEntity could look like:
```
@Entity
private static class EmbeddedEntity {
    @Id Long id;
    boolean foo;
}
```

To prove that this is an issue, I also created a unit test:
https://github.com/msomhorst/objectify/commit/2db78802d9094f5563153644bae59af36c1159fa#diff-39a25c5b3e41085a405cca6d46c25f6fR124

**Comment by msomhorst** [comment]
My question would be: is this a bug, or a feature?

**Comment by stickfigure** [comment]
So this is actually a feature, not a bug... although I agree it is a little confusing.

Take a look at this test:
https://github.com/objectify/objectify/blob/master/src/test/java/com/googlecode/objectify/test/DatastoreTests.java#L102

The behavior of Google's SDK is that if you don't index an embedded Entity node, all of its children will be unindexed - even if you explicitly ask them to be indexed. This means Objectify can only index deeper properties by keeping all the container nodes indexed.

Technically I suppose Objectify could figure out that nothing deeper has been indexed and unindex the container nodes... but there doesn't seem to be much point. AFAICT, there's zero cost associated with "indexing" an intermediate node - it doesn't create any stored data. Only leaf property values are actually stored in the index.

TL;DR: Setting a container as indexed doesn't actually create an index, it really means "allow child properties to be indexed" and we have problems if we turn that off.

**Comment by msomhorst** [comment]
Great! If it doesn't actually cost any storage to "index" the intermediate node, there is indeed no point in excluding it from indexing. Thanks for the answer!
